Documentation/debugging/windows-instructions.md

```diff
@@ -0,0 +1,93 @@
+Debugging CoreFX on Windows
+==========================
+
+You can Debug .NET Core via Visual Studio or WinDBG.
+
+For Visual Studio debugging, follow the instructions at [Debugging tests in Visual Studio](https://github.com/dotnet/corefx/blob/master/Documentation/building/windows-instructions.md) to run and debug tests.
+For bugs that cannot be reproduced within Visual Studio (certain low-probability race conditions, SafeHandle life-time problems, etc) you will need to use WinDBG.
+
+## Required Software
+
+WinDBG free download:
+
+* [WDK and WinDBG downloads](https://msdn.microsoft.com/en-us/windows/hardware/hh852365.aspx)
+
+Note: You can select the Standalone Debugging Tools for Windows or download only WinDBG via the WDK.
+
+## Prerequisites to Debugging with WinDBG
+
+1. Build the entire repository. This ensures that all packages are downloaded and that you have up-to-date symbols.
+
+2. Find the path where the debugging engine (SOS) is located.
+
+Assuming that your repo is at `C:\corefx` this should be located at:
+
+```
+C:\corefx\packages\runtime.win7-x64.Microsoft.NETCore.Runtime.CoreCLR\<version>\tools\sos.dll   
+```
+
+3. Install WinDBG as post-mortem debugger
+As Administrator:
+
+```
+windbg -I
+```
+
+You may need to do this for both x64 and x86 versions. 
+Any application that crashes should now automatically start a WinDBG session.
+
+## Debugging tests
+To run a single test from command line:
+
+* Locate the test binary folder based on the CSPROJ name. 
+
+For example: `src\System.Net.Sockets\tests\Functional\System.Net.Sockets.Tests.csproj` will build and output binaries at  `bin\tests\Windows_NT.AnyCPU.Debug\System.Net.Sockets.Tests\dnxcore50`.
+ 
+* Execute the test
+
+Assuming that your repo is at `C:\corefx`:
+
+```
+cd C:\corefx\bin\tests\Windows_NT.AnyCPU.Debug\System.Net.Sockets.Tests\dnxcore50
+C:\corefx\bin\tests\Windows_NT.AnyCPU.Debug\System.Net.Sockets.Tests\dnxcore50\CoreRun.exe xunit.console.netcore.exe System.Net.Sockets.Tests.dll -xml testResults.xml -notrait category=nonwindowstests -notrait category=OuterLoop -notrait category=failing
+```
+
+* If the test crashes or encounteres a `Debugger.Launch()` method call, WinDBG will automatically start and attach to the `CoreRun.exe` process 
+
```

Comment:
There is no child process, xunit.console.netcore.exe is just a regular .NET executable assembly that CoreCLR executes, and the runner executes all of the tests in-process.